src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -691,13 +689,21 @@ private void PopulateSessionProperties(OpenIdConnectMessage message, Authenticat
         protected virtual async Task<OpenIdConnectMessage> RedeemAuthorizationCodeAsync(OpenIdConnectMessage tokenEndpointRequest)
         {
             Logger.RedeemingCodeForTokens();
+
             var requestMessage = new HttpRequestMessage(HttpMethod.Post, _configuration.TokenEndpoint);
             requestMessage.Content = new FormUrlEncodedContent(tokenEndpointRequest.Parameters);
+
             var responseMessage = await Backchannel.SendAsync(requestMessage);
-            responseMessage.EnsureSuccessStatusCode();
-            var tokenResonse = await responseMessage.Content.ReadAsStringAsync();
-            var jsonTokenResponse = JObject.Parse(tokenResonse);
-            return new OpenIdConnectMessage(jsonTokenResponse);
+            var message = new OpenIdConnectMessage(await responseMessage.Content.ReadAsStringAsync());
+
+            if (string.IsNullOrEmpty(message.Error))
```

Comment:
Sure, `OpenIdConnectMessage` will likely throw an exception if it cannot parse the response body. My remark was more about the exception message: if a server returns an empty 500 response, will the message be clear enough about the fact the server was faulty?